app/services/post_webhook.rb

```diff
@@ -0,0 +1,72 @@
+require 'base64'
+require 'rest-client'
+
+class PostWebhook
+  USER_AGENT = 'Redhat Insights Portal v1.0.0'.freeze
+  SIGNATURE_KEY = 'X-Service-Portal-Signature'.freeze
+
+  def initialize(options)
+    @options = options
+    validate
+    default_headers
+    authorization
+  end
+
+  def process(json)
+    hmac(json)
+    post(json)
+  end
+
+  private
+
+  def validate
+    raise ArgumentError, "url not specified" if @options['url'].blank?
+    validate_auth
+  end
+
+  def validate_auth
+    case @options['authentication']
+    when 'oauth'
+      raise ArgumentError, 'bearer token not specified' if @options['token'].blank?
+    when 'basic'
+      raise ArgumentError, 'username not specified' if @options['username'].blank?
+      raise ArgumentError, "password not specified" if @options['password'].blank?
+    end
+  end
+
+  def default_headers
+    @headers = {
+      'User-Agent'   => USER_AGENT,
+      'Content-Type' => "application/json",
+      'Accept'       => "application/json"
+    }
+  end
+
+  def authorization
+    case @options['authentication']
+    when 'basic'
+      @headers['Authorization'] = "Basic " + Base64.encode64("#{@options['username']}:#{@options['password']}")
```

Comment:
Possibly change to `strict_encode64`

Reply:
@mkanoor Can you also fix this one?  While you're at it, prefer interpolation over string concatenation.